### DISCUSSION TITLE
u-boot  can not to be run

### DISCUSSION THREAD

**Discussion by yeliangchen2** [polarfire-soc/polarfire-soc-documentation#143] [HSS and U-Boot] [answered]
1. hss version:
![image](https://user-images.githubusercontent.com/16462894/179909576-f6187653-2637-401d-8925-8fa84486ecec.png)

2. log (infos of all hards output to uart0):
![image](https://user-images.githubusercontent.com/16462894/179909950-02d1341d-82e4-4d53-9185-796cbd65881a.png)

3.uboot dtb change the uart1->uart0:
![image](https://user-images.githubusercontent.com/16462894/179910538-c70f7e31-ba10-4929-9e9b-2d98744ef031.png)
![image](https://user-images.githubusercontent.com/16462894/179913115-76dcd4ad-1383-426f-b213-429feb861192.png)

4.read payload.bin ->0x8000 0000
5.copy 0x8000 07b8 -> 0x10 0000 0000 (0x7b8 is an offset of my uboot.bin )
6. I check the data from the 0x10 0000 0000 is right .(the same with my local uboot.bin)
7. could you provide me the register address of checking for all harts status .
8.or other helps, thanks

**Comment by hughbreslin** [comment]
Hi @yeliangchen2 

The HSS and U-Boot may be fighting over the UART as they're sharing it. The register you're asking for in 7 doesn't exist unfortunately, what you could do is connect the debugger without downloading code or resetting the hart and see if a hart has trapped and also check the `mcause` value to see why it has trapped. 

Another suggestion (you could do this as well as or instead of the above) is to enable the `UART_SURRENDER` config option in the HSS. To do this you will need to disable `SERVICE_TINYCLI_REGISTER`. This will get the HSS to give up UART0 when it hands over to U-Boot.

E.g from the command line or menu config, in `Services` deselect `TinyCLI support`:
![image](https://user-images.githubusercontent.com/44862881/180186132-60f512a1-fefb-4eba-b438-d5c632024f0e.png)

and then in `General Configuration Options` -> `Miscellaneous` enable `HSS UART Surrender`:
![image](https://user-images.githubusercontent.com/44862881/180186421-382132d8-e219-447e-bccf-0c33a84e16f1.png)

**Reply by xiyanni** [reply]
Below is update from FAE of NR. Please pay attention to it and give a hand. Thanks!

1.	Since the peripheral interfaces are not exactly the same as ICICLE kit and FIC interfaces are also different, they have to modify the HSS code. We can’t guarantee the modification is good. I just listed the differences.
a.	LPDDR4->DDR3L     MSS DDR memory space is 256Mb
b.	SD card 8 wires to 4 wires.
c.	Only one UART physically.
d.	Delete the some FIC address apace. The gray items. It means the FPGA design is updated.  Questions here: if HSS need to visit the memory space in the gray items?

2.	After the modification,  HSS can copy Payload.bin to MDDR successfully. but can’t boot U-boot + BM. The U-boot boot failure at U54_1 and the BM boot successful at U54_4. 

3.	NR did some further debug, They change Yaml file to boot BM + BM example, HSS can boot BM at U54_1 and boot another BM at U54_4 simultaneously. Attached files are customer’s Yaml and boot log.   By the way, the customer only have UART0 for debug purpose. So, the log file contains E51 U54_1 and U54_4 UART output.

4.	Based on the test, looks like there are some issues with the UBOOT, do they need to modify the UBOOT to adapt their hardware?

5.	NR’s modified HSS code is uploaded to the case.

**Comment by xiyanni** [comment, marked as answer]
This issue has been solved by the customer. Due to the customer change the run address from 0x1000200000 to other address in Yaml file. Which cause run address different with link address. So, the u-boot run failure.
